apps/web/src/components/providers/OptInProvider.tsx

```diff
@@ -20,32 +19,29 @@ export const OptInProvider = ({ children }: { children: React.ReactNode }) => {
 };
 
 export const _OptInProvider = (props: PropsWithChildren) => {
-  const navigate = useNavigate();
   const { children } = props;
-  const { status, isLoaded } = useNewDashboardOptIn();
+  const { status, isLoaded, redirectToNewDashboard } = useNewDashboardOptIn();
 
   useEffect(() => {
-    const dashboardV2HostName = window.location.hostname.includes('dashboard-v2');
-
-    if (isLoaded && status === NewDashboardOptInStatusEnum.OPTED_IN && dashboardV2HostName) {
+    if (isLoaded && status === NewDashboardOptInStatusEnum.OPTED_IN) {
       const currentRoute = window.location.pathname.replace('/legacy', '');
 
       /**
        * if equivalent of current route (incl. subroutes) exits in new dashboard, redirect to it
        * - /legacy/workflows -> /workflows
        * - /legacy/workflows/edit/123 -> /workflows
        */
-      if (NEW_DASHBOARD_ROUTES.some((route) => currentRoute.includes(route))) {
+      if (ROUTES_THAT_REDIRECT_TO_DASHBOARD.some((route) => currentRoute.includes(route))) {
         /**
          * TODO: in order to redirect to the same route, we need to translate the
          * "dev_env_<id>" or wf/step slugs to legacy environment id and vice-versa
          *
          * note: /legacy is part of public URL, so we can't navigate() outside of that
          */
-        window.location.href = window.location.origin;
+        redirectToNewDashboard();
```

Comment:
when the user is opted-in and is in one of the `/workflows` routes, then redirect to the new dashboard